Parsers/ASimNetworkSession/Parsers/vimNetworkSessionVMwareCarbonBlackCloud.yaml

```diff
@@ -0,0 +1,205 @@
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: VMware Carbon Black Cloud documentation 
+  Link: https://developer.carbonblack.com/reference/carbon-black-cloud/
+        https://developer.carbonblack.com/reference/cb-threathunter/latest/event-search-fields/
+Description: |
+  This ASIM parser supports normalizing VMware Carbon Black Cloud logs to the ASIM Registry Event normalized schema. VMware Carbon Black Cloud events are captured through VMware Carbon Black Cloud data connector which ingests Carbon Black Audit, Notification and Event data into Microsoft Sentinel through the REST API.
+ParserName: vimNetworkSessionVMwareCarbonBlackCloud
+EquivalentBuiltInParser: _Im_NetworkSession_VMwareCarbonBlackCloud
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstportnumber
+    Type: int
+    Default: int(null)
+  - Name: dvcaction
+    Type: dynamic
+    Default: dynamic([])
+  - Name: hostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let NetworkProtocolLookup = datatable (netconn_protocol_s: string, NetworkProtocol: string)
+      [
+      "PROTO_TCP", "TCP",
+      "PROTO_UDP", "UDP"
+  ];
+  let DvcActionLookup = datatable (sensor_action_s: string, DvcAction: string)
+      [
+      "ACTION_ALLOW", "Allow",
+      "ACTION_SUSPEND", "Drop",
+      "ACTION_TERMINATE", "Drop",
+      "ACTION_BREAK", "Drop",
+      "ACTION_BLOCK", "Deny"
+  ];
+  let parser=(
+      disabled: bool=false, 
+      starttime: datetime=datetime(null), 
+      endtime: datetime=datetime(null), 
+      eventresult: string='*', 
+      srcipaddr_has_any_prefix: dynamic=dynamic([]),
+      dstipaddr_has_any_prefix: dynamic=dynamic([]),
+      ipaddr_has_any_prefix: dynamic=dynamic([]), 
+      hostname_has_any: dynamic=dynamic([]),
+      dstportnumber: int=int(null),
+      dvcaction: dynamic=dynamic([])
+      ) {
+      let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      CarbonBlackEvents_CL
+      | where not(disabled)
+          and (isnull(starttime) or TimeGenerated >= starttime)
+          and (isnull(endtime) or TimeGenerated <= endtime)
+          and array_length(hostname_has_any) == 0
+          and eventType_s == "endpoint.event.netconn"
+          and (isnull(dstportnumber) or toint(remote_port_d) == dstportnumber)
+      | lookup NetworkProtocolLookup on netconn_protocol_s
+      | lookup DvcActionLookup on sensor_action_s
```

Comment:
can you check why with EventResult Failure. We are getting DvcAction Allow?
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/f2280259-779c-4d71-a6fa-3f5e94fc310a)

Reply:
If the source device does not provide an event result, EventResult should be based on the value of [DvcAction](https://learn.microsoft.com/en-us/azure/sentinel/normalization-schema-network#dvcaction). If [DvcAction](https://learn.microsoft.com/en-us/azure/sentinel/normalization-schema-network#dvcaction) is Deny, Drop, Drop ICMP, Reset, Reset Source, or Reset Destination
, EventResult should be Failure. Otherwise, EventResult should be Success.

Reply:
In the case of EventOriginalType = ACTION_CONNECTION_CREATE_FAILED, we have set EventResult as Failure directly as the event itself suggests the failure. For other types of events we have decided EventResult based on DvcAction like Success for ACTION_ALLOW else Failure.

Reply:
@jayeshprajapaticrest @niralishah-crest 
You meant we could have DvcAction as Allow and EventResult as Failure?

Reply:
Yes

Reply:
@jayeshprajapaticrest @niralishah-crest 
can you please explain this scenario where EventResult and DvcAction would be contradict each other?

Reply:
When there is an event called "ACTION_CONNECTION_CREATE_FAILED" occured, it indicates that there is a failure  for establishing a connection which means EventResult is Failure. For DvcAction, we are considering field called sensor_action_s and for these types of events we are getting sensor_action = ACTION_ALLOW which means VMware has allowed it. That's why we have mapped DvcAction as Allow and EventResult as Failure.